### PR TITLE
Make sure that Davix file is only closed once

### PR DESCRIPTION
**Pull request by Dr15Jones**
Telling davix to close the same file multiple times is causing a seg fault. By releasing the underlying file after closing, we avoid that problem.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

File: Utilities/DavixAdaptor/src/DavixFile.cc [hunk 1]
```diff
@@ -33,8 +33,9 @@ DavixFile::~DavixFile(void) {
 
 void DavixFile::close(void) {
   if (m_davixPosix && m_fd) {
+    auto davixPosix = std::move(m_davixPosix);
     DavixError *err = nullptr;
-    m_davixPosix->close(m_fd, &err);
+    davixPosix->close(m_fd, &err);
```
Comment (bbockelm) [inline Utilities/DavixAdaptor/src/DavixFile.cc]: How about also setting `m_fd` to `nullptr`?

```suggestion
    davixPosix->close(m_fd, &err);
    m_fd = nullptr;
```

**Review by bbockelm — COMMENTED** [review]
Looks fine, although might be better to also NULL-out `m_fd`.

File: Utilities/DavixAdaptor/src/DavixFile.cc (hunk 1 above)
Comment (Dr15Jones) [reply Utilities/DavixAdaptor/src/DavixFile.cc]: Sure, although to use `m_fd` one first has to dereference `m_davixPosix`.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

CMSSW Code rules violation is unrelated to this PR (DDCMS already discussed with @ianna)

**Pull request merged**